### PR TITLE
Adds tests for the omicli

### PR DESCRIPTION
**Pull request by c64cosmin**
Adds tests for the cli only.

### CODE REVIEW

File: provider/GNUmakefile [hunk 1]
```diff
@@ -56,9 +56,6 @@ $(OBJECTS) : | $(OBJ_PATH)
 
 CPPFLAGS+=$(INCLUDES)
 
-PRINT_BOOKENDS?=0
-CPPFLAGS+=-DPRINT_BOOKENDS=$(PRINT_BOOKENDS)
```
Comment (JumpingYang001) [inline provider/GNUmakefile:60]: you remove DPRINT_BOOKENDS, how can user show the debug info? is there other ways to enable this?

**Review by JumpingYang001 — APPROVED** [review]
1 comment.

**Comment by JumpingYang001** [comment]
BTW, please squash into 1 commit, thanks.

File: provider/GNUmakefile (hunk 1 above)
Comment (c64cosmin) [reply provider/GNUmakefile:60]: Yes.
export CPPFLAGS="-DPRINT_BOOKENDS"
It should get documented!

**Comment by c64cosmin** [comment]
I'm working on a revert to fix the issues with this branch. Also will take in consideration the failing of the XYZ_Number test.

**Pull request merged**